services/motion/builtin/move_request.go

```diff
@@ -72,7 +75,9 @@ func (mr *moveRequest) execute(ctx context.Context, waypoints [][]referenceframe
 
 	// the plan has been fully executed so check to see if the GeoPoint we are at is close enough to the goal.
 	deviated, err := mr.deviatedFromPlan(ctx, waypoints, len(waypoints)-1)
-	if err != nil {
+	if errors.Is(err, context.Canceled) {
+		return moveResponse{}
```

Comment:
It's not clear to me that this is the point where we want to be making this decision. If we want sometimes-different behavior based on certain error messages, my inclination would be to push that as high up the stack as absolutely possible, and have as much of the stack as possible return the actual error if encountered.